### PR TITLE
Generate dependencies on commit

### PR DESCRIPTION
**Pull request by ilesev-vmw**
Signed-off-by: Iliyan Lesev <ilesev@vmware.com>

### CODE REVIEW

**Comment by Astardjiev** [comment]
STAR-3245

File: .github/workflows/generate-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const path = require('path');
+
+const ignoredFolders = ['.github', '.gradle', '.idea', '.git', 'node_modules', 'lib', 'src', 'tests'];
+const generatedFileName = 'provenance.json';
+const projectName = 'vcd-ext-sdk/';
+
+async function generateFileHelper(currentDir, components) {
```
Comment (stoychevl) [inline .github/workflows/generate-dependencies.js:8]: Can we avoid function names such as Helper, Util, Common etc. ? If the name of the function cannot be descriptive enough, let's add a code doc above.

File: .github/workflows/generate-dependencies.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    await generateFileHelper(filePath, components);
+                }
+            }
+        }
+    } catch (e) {
+        console.error("Could not generate dependencies for project.", e);
+    }
+}
+
+async function readFile(filePath) {
```
Comment (stoychevl) [inline .github/workflows/generate-dependencies.js:33]: If I call this function without taking a look at it, I would expect that it returns a raw file content.

File: .github/workflows/generate-dependencies.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        version: version,
+        dependencies: []
+    };
+
+    extract(packageLockJson.dependencies, component);
+
+    components[packageLockJson.name] = component;
+}
+
+function extract(dependencies, component) {
```
Comment (stoychevl) [inline .github/workflows/generate-dependencies.js:50]: Perhaps also give a more descriptive naming such as extractDependencyInfo?

File: .github/workflows/generate-dependencies.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                }
+            }
+        }
+    } catch (e) {
+        console.error("Could not generate dependencies for project.", e);
+    }
+}
+
+async function readFile(filePath) {
+    const fileContent = await fs.readFileSync(filePath);
```
Comment (stoychevl) [inline .github/workflows/generate-dependencies.js:34]: Why are you awaiting a sync function? If you don't, the function itself shouldn't be async, which itself is used in a few places with async await. Generally speaking, if you don't have the need of parallel promise handling, try to stick with synchronous implementation.

**Review by stoychevl — APPROVED** [review]

**Pull request merged**